src/lib/emitters.rs

```diff
@@ -383,11 +399,138 @@ impl RenderDotfile for MappingStore<String> {
                 MappingType::RECOVERY => "[style=dotted, color=green, ",
                 MappingType::EDIT => "[style=dotted, color=indigo, ",
             };
-            line = format!("\t{{ rank=same FROM{} -> TO{}{}{}]; }}\n",
-                           from.id(),
-                           to.id(),
-                           attrs,
-                           common);
+            line = format!(
+                "\t{{ rank=same FROM{} -> TO{}{}{}]; }}\n",
+                from.id(),
+                to.id(),
+                attrs,
+                common
+            );
+            digraph.push(line);
+        }
+        digraph.push(String::from("}\n"));
+        // Write dotfile to stream.
+        buffer.write_all(digraph.join("").as_bytes())
+    }
+}
+
+
+
+impl RenderDotfile for MappingStore2<String> {
+    // Because the `dot` crate cannot add arbitrary attributes to nodes and
+    // edges, or create subgraphs and clusters, we render mapping stores to
+    // a dot digraph manually. This produces output similar to the Figure 1
+    // of Falleri et al. (2014).
```

Comment:
I'm not sure that "...similar to Figure 1 of Falleri et al. (2014)" is true for your dotfiles? If not let's kill that bit of the comment, or replace it with something more sensible.